src/devices/RGBLedMatrix/RgbLedMatrix.cs

```diff
@@ -0,0 +1,641 @@
+        /// <param name="bkR">text background red color</param>
+        /// <param name="bkG">text background green color</param>
+        /// <param name="bkB">text background blue color</param>
+        /// <param name="backBuffer">true if want use back buffer, false otherwise</param>
+        public void DrawText(int x, int y, string text, BdfFont font, byte textR, byte textG, byte textB, byte bkR, byte bkG, byte bkB, bool backBuffer = false)
+        {
+            DrawText(x, y, text.AsSpan(), font, textR, textG, textB, bkR, bkG, bkB, backBuffer);
+        }
+
+        /// <summary>
+        /// Draw a Circle on the display
+        /// </summary>
+        /// <param name="xCenter">x coordinate of the center</param>
+        /// <param name="yCenter">y coordinate of the center</param>
+        /// <param name="radius">radius length</param>
+        /// <param name="red">red color of circle arc</param>
+        /// <param name="green">green color of circle arc</param>
+        /// <param name="blue">blue color of circle arc</param>
+        /// <param name="backBuffer">true if want use back buffer, false otherwise</param>
+        public void DrawCircle(int xCenter, int yCenter, int radius, byte red, byte green, byte blue, bool backBuffer = false)
+        {
+            byte [] buffer = backBuffer ? _colorsBackBuffer : _colorsBuffer;
+
+            for (double angle = 0.0; angle < 6.2832; angle += 1.0 / radius)
+            {
+                SetPixel((int) Math.Round(xCenter + radius * Math.Cos(angle)), (int) Math.Round(yCenter + radius * Math.Sin(angle)), red, green, blue, buffer);
+            }
+        }
+
+        private void DrawChar(int x, int y, char c, BdfFont font, byte textR, byte textG, byte textB, byte bkR, byte bkG, byte bkB, byte [] buffer)
+        {
+            int hightToDraw = Math.Min(Height - y, font.Height);
+            int firstColumnToDraw = x < 0 ? Math.Abs(x) : 0;
+            int lastColumnToDraw  = x + font.Width > Width ? Width - x : font.Width;
+
+            font.GetCharData(c, out ReadOnlySpan<ushort> charData);
+
+            int b = 8 * (sizeof(ushort) - (int) Math.Ceiling(((double)font.Width) / 8)) + firstColumnToDraw;
+
+            for (int j = firstColumnToDraw; j < lastColumnToDraw; j++)
+            {
+                for (int i = 0; i < hightToDraw; i++)
+                {
+                    int value = charData[i] << (b + j - firstColumnToDraw);
+
+                    if ((value & 0x8000) != 0)
+                        SetPixel(x + j, y + i, textR, textG, textB, buffer);
+                    else
+                        SetPixel(x + j, y + i, bkR, bkG, bkB, buffer);
+                }
+            }
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private void Sleep(long duration)
+        {
+            long startTicks = Stopwatch.GetTimestamp();
+            while (Stopwatch.GetTimestamp() - startTicks < duration) { }
```

Comment:
could this use DelayHelpers?

Reply:
I would wait to make this common because if I need to tweak it more it will be specific to this device and not common. let's watch if anyone else will need similar functionality.